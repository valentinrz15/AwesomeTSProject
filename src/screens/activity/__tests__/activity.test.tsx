import React from 'react';
import renderer from 'react-test-renderer';
import { Activity } from '../index';

describe('activity index test', () => {
  it('should render activity header titles', () => {
    const { toJSON } = renderer.create(<Activity />);

    expect(toJSON()).toMatchSnapshot();
  });
});
