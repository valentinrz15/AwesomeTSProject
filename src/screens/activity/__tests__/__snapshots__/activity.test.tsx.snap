// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`activity index test should render activity header titles 1`] = `
Array [
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#FFF",
        "flexDirection": "row",
        "justifyContent": "space-between",
        "paddingHorizontal": 20,
        "paddingVertical": 10,
      }
    }
  >
    <View
      accessible={true}
      collapsable={false}
      focusable={false}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "backgroundColor": "#000",
          "borderColor": "#ccc",
          "borderRadius": 5,
          "borderWidth": 0.5,
          "opacity": 1,
          "paddingHorizontal": 5,
          "paddingVertical": 3,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#FFF",
            "fontSize": 18,
          }
        }
      >
        Close
      </Text>
    </View>
    <View
      accessible={true}
      collapsable={false}
      focusable={false}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "backgroundColor": "#000",
          "borderColor": "#ccc",
          "borderRadius": 5,
          "borderWidth": 0.5,
          "opacity": 1,
          "paddingHorizontal": 5,
          "paddingVertical": 3,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#FFF",
            "fontSize": 18,
          }
        }
      >
        Filters
      </Text>
    </View>
  </View>,
  <RCTScrollView
    contentContainerStyle={
      Object {
        "borderWidth": 1,
        "justifyContent": "center",
        "paddingHorizontal": 20,
      }
    }
    data={
      Array [
        Object {
          "content": Array [
            Object {
              "id": "1",
              "title": "Content 1",
            },
            Object {
              "id": "2",
              "title": "Content 2",
            },
            Object {
              "id": "3",
              "title": "Content 3",
            },
            Object {
              "id": "4",
              "title": "Content 4",
            },
            Object {
              "id": "5",
              "title": "Content 5",
            },
            Object {
              "id": "6",
              "title": "Content 6",
            },
          ],
          "data": Array [
            Object {
              "id": "1",
              "title": "Content 1",
            },
          ],
          "header": Array [
            Object {
              "id": "1",
              "title": "Header 1",
            },
            Object {
              "id": "2",
              "title": "Header 2",
            },
            Object {
              "id": "3",
              "title": "Header 3",
            },
            Object {
              "id": "4",
              "title": "Header 4",
            },
            Object {
              "id": "5",
              "title": "Header 5",
            },
            Object {
              "id": "6",
              "title": "Header 6",
            },
            Object {
              "id": "7",
              "title": "Header 7",
            },
          ],
        },
      ]
    }
    getItem={[Function]}
    getItemCount={[Function]}
    keyExtractor={[Function]}
    onContentSizeChange={[Function]}
    onLayout={[Function]}
    onMomentumScrollBegin={[Function]}
    onMomentumScrollEnd={[Function]}
    onScroll={[Function]}
    onScrollBeginDrag={[Function]}
    onScrollEndDrag={[Function]}
    renderItem={[Function]}
    scrollEventThrottle={50}
    stickyHeaderHiddenOnScroll={false}
    stickyHeaderIndices={Array []}
    style={
      Object {
        "backgroundColor": "#FFF",
        "flex": 1,
      }
    }
    testID="activity-index-section-list"
  >
    <View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <Text
          style={
            Object {
              "color": "#000",
              "fontSize": 22,
            }
          }
        >
          Section 1
        </Text>
        <View
          contentContainerStyle={
            Object {
              "paddingHorizontal": 20,
              "width": 375,
            }
          }
          data={
            Array [
              Object {
                "id": "1",
                "title": "Header 1",
              },
              Object {
                "id": "2",
                "title": "Header 2",
              },
              Object {
                "id": "3",
                "title": "Header 3",
              },
              Object {
                "id": "4",
                "title": "Header 4",
              },
              Object {
                "id": "5",
                "title": "Header 5",
              },
              Object {
                "id": "6",
                "title": "Header 6",
              },
              Object {
                "id": "7",
                "title": "Header 7",
              },
            ]
          }
          getItem={[Function]}
          getItemCount={[Function]}
          keyExtractor={[Function]}
          onContentSizeChange={[Function]}
          onLayout={[Function]}
          onMomentumScrollBegin={[Function]}
          onMomentumScrollEnd={[Function]}
          onScroll={[Function]}
          onScrollBeginDrag={[Function]}
          onScrollEndDrag={[Function]}
          removeClippedSubviews={false}
          renderItem={[Function]}
          scrollEnabled={true}
          scrollEventThrottle={50}
          showsHorizontalScrollIndicator={false}
          showsVerticalScrollIndicator={false}
          stickyHeaderIndices={Array []}
          testID="activity-activity-list-index-flat-list"
          viewabilityConfigCallbackPairs={Array []}
        >
          <View
            onLayout={[Function]}
            style={null}
          >
            <View
              style={
                Object {
                  "flexDirection": "row",
                }
              }
            >
              <View
                style={
                  Object {
                    "flexDirection": "row",
                  }
                }
                testID="activity-activity-list-content-container"
              >
                <View
                  accessible={true}
                  collapsable={false}
                  focusable={false}
                  onClick={[Function]}
                  onResponderGrant={[Function]}
                  onResponderMove={[Function]}
                  onResponderRelease={[Function]}
                  onResponderTerminate={[Function]}
                  onResponderTerminationRequest={[Function]}
                  onStartShouldSetResponder={[Function]}
                  style={
                    Object {
                      "borderRadius": 10,
                      "borderWidth": 1,
                      "marginHorizontal": 5,
                      "marginVertical": 10,
                      "opacity": 1,
                      "padding": 5,
                    }
                  }
                  testID="activity-activity-list-content-button"
                >
                  <Text>
                    Header 1
                  </Text>
                </View>
              </View>
              <View
                style={
                  Object {
                    "flexDirection": "row",
                  }
                }
                testID="activity-activity-list-content-container"
              >
                <View
                  accessible={true}
                  collapsable={false}
                  focusable={false}
                  onClick={[Function]}
                  onResponderGrant={[Function]}
                  onResponderMove={[Function]}
                  onResponderRelease={[Function]}
                  onResponderTerminate={[Function]}
                  onResponderTerminationRequest={[Function]}
                  onStartShouldSetResponder={[Function]}
                  style={
                    Object {
                      "borderRadius": 10,
                      "borderWidth": 1,
                      "marginHorizontal": 5,
                      "marginVertical": 10,
                      "opacity": 1,
                      "padding": 5,
                    }
                  }
                  testID="activity-activity-list-content-button"
                >
                  <Text>
                    Header 2
                  </Text>
                </View>
              </View>
              <View
                style={
                  Object {
                    "flexDirection": "row",
                  }
                }
                testID="activity-activity-list-content-container"
              >
                <View
                  accessible={true}
                  collapsable={false}
                  focusable={false}
                  onClick={[Function]}
                  onResponderGrant={[Function]}
                  onResponderMove={[Function]}
                  onResponderRelease={[Function]}
                  onResponderTerminate={[Function]}
                  onResponderTerminationRequest={[Function]}
                  onStartShouldSetResponder={[Function]}
                  style={
                    Object {
                      "borderRadius": 10,
                      "borderWidth": 1,
                      "marginHorizontal": 5,
                      "marginVertical": 10,
                      "opacity": 1,
                      "padding": 5,
                    }
                  }
                  testID="activity-activity-list-content-button"
                >
                  <Text>
                    Header 3
                  </Text>
                </View>
              </View>
              <View
                style={
                  Object {
                    "flexDirection": "row",
                  }
                }
                testID="activity-activity-list-content-container"
              >
                <View
                  accessible={true}
                  collapsable={false}
                  focusable={false}
                  onClick={[Function]}
                  onResponderGrant={[Function]}
                  onResponderMove={[Function]}
                  onResponderRelease={[Function]}
                  onResponderTerminate={[Function]}
                  onResponderTerminationRequest={[Function]}
                  onStartShouldSetResponder={[Function]}
                  style={
                    Object {
                      "borderRadius": 10,
                      "borderWidth": 1,
                      "marginHorizontal": 5,
                      "marginVertical": 10,
                      "opacity": 1,
                      "padding": 5,
                    }
                  }
                  testID="activity-activity-list-content-button"
                >
                  <Text>
                    Header 4
                  </Text>
                </View>
              </View>
            </View>
          </View>
          <View
            onLayout={[Function]}
            style={null}
          >
            <View
              style={
                Object {
                  "flexDirection": "row",
                }
              }
            >
              <View
                style={
                  Object {
                    "flexDirection": "row",
                  }
                }
                testID="activity-activity-list-content-container"
              >
                <View
                  accessible={true}
                  collapsable={false}
                  focusable={false}
                  onClick={[Function]}
                  onResponderGrant={[Function]}
                  onResponderMove={[Function]}
                  onResponderRelease={[Function]}
                  onResponderTerminate={[Function]}
                  onResponderTerminationRequest={[Function]}
                  onStartShouldSetResponder={[Function]}
                  style={
                    Object {
                      "borderRadius": 10,
                      "borderWidth": 1,
                      "marginHorizontal": 5,
                      "marginVertical": 10,
                      "opacity": 1,
                      "padding": 5,
                    }
                  }
                  testID="activity-activity-list-content-button"
                >
                  <Text>
                    Header 5
                  </Text>
                </View>
              </View>
              <View
                style={
                  Object {
                    "flexDirection": "row",
                  }
                }
                testID="activity-activity-list-content-container"
              >
                <View
                  accessible={true}
                  collapsable={false}
                  focusable={false}
                  onClick={[Function]}
                  onResponderGrant={[Function]}
                  onResponderMove={[Function]}
                  onResponderRelease={[Function]}
                  onResponderTerminate={[Function]}
                  onResponderTerminationRequest={[Function]}
                  onStartShouldSetResponder={[Function]}
                  style={
                    Object {
                      "borderRadius": 10,
                      "borderWidth": 1,
                      "marginHorizontal": 5,
                      "marginVertical": 10,
                      "opacity": 1,
                      "padding": 5,
                    }
                  }
                  testID="activity-activity-list-content-button"
                >
                  <Text>
                    Header 6
                  </Text>
                </View>
              </View>
              <View
                style={
                  Object {
                    "flexDirection": "row",
                  }
                }
                testID="activity-activity-list-content-container"
              >
                <View
                  accessible={true}
                  collapsable={false}
                  focusable={false}
                  onClick={[Function]}
                  onResponderGrant={[Function]}
                  onResponderMove={[Function]}
                  onResponderRelease={[Function]}
                  onResponderTerminate={[Function]}
                  onResponderTerminationRequest={[Function]}
                  onStartShouldSetResponder={[Function]}
                  style={
                    Object {
                      "borderRadius": 10,
                      "borderWidth": 1,
                      "marginHorizontal": 5,
                      "marginVertical": 10,
                      "opacity": 1,
                      "padding": 5,
                    }
                  }
                  testID="activity-activity-list-content-button"
                >
                  <Text>
                    Header 7
                  </Text>
                </View>
              </View>
            </View>
          </View>
        </View>
      </View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <Text
          style={
            Object {
              "color": "#000",
              "fontSize": 22,
            }
          }
        >
          Section 2
        </Text>
        <View
          contentContainerStyle={
            Object {
              "paddingHorizontal": 20,
              "width": 375,
            }
          }
          data={
            Array [
              Object {
                "id": "1",
                "title": "Content 1",
              },
            ]
          }
          getItem={[Function]}
          getItemCount={[Function]}
          keyExtractor={[Function]}
          onContentSizeChange={[Function]}
          onLayout={[Function]}
          onMomentumScrollBegin={[Function]}
          onMomentumScrollEnd={[Function]}
          onScroll={[Function]}
          onScrollBeginDrag={[Function]}
          onScrollEndDrag={[Function]}
          removeClippedSubviews={false}
          renderItem={[Function]}
          scrollEnabled={true}
          scrollEventThrottle={50}
          showsHorizontalScrollIndicator={false}
          showsVerticalScrollIndicator={false}
          stickyHeaderIndices={Array []}
          testID="activity-activity-list-index-flat-list"
          viewabilityConfigCallbackPairs={Array []}
        >
          <View
            onLayout={[Function]}
            style={null}
          >
            <View
              style={
                Object {
                  "flexDirection": "row",
                }
              }
              testID="activity-activity-list-content-container"
            >
              <View
                accessible={true}
                collapsable={false}
                focusable={false}
                onClick={[Function]}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  Object {
                    "borderRadius": 10,
                    "borderWidth": 1,
                    "marginHorizontal": 5,
                    "marginVertical": 10,
                    "opacity": 1,
                    "padding": 5,
                  }
                }
                testID="activity-activity-list-content-button"
              >
                <Text>
                  Content 1
                </Text>
              </View>
            </View>
          </View>
        </View>
      </View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <Text
          style={
            Object {
              "color": "#000",
              "fontSize": 22,
            }
          }
        >
          Section 3
        </Text>
        <View
          contentContainerStyle={
            Object {
              "paddingHorizontal": 20,
              "width": 375,
            }
          }
          data={
            Array [
              Object {
                "id": "1",
                "title": "Content 1",
              },
              Object {
                "id": "2",
                "title": "Content 2",
              },
              Object {
                "id": "3",
                "title": "Content 3",
              },
              Object {
                "id": "4",
                "title": "Content 4",
              },
              Object {
                "id": "5",
                "title": "Content 5",
              },
              Object {
                "id": "6",
                "title": "Content 6",
              },
            ]
          }
          getItem={[Function]}
          getItemCount={[Function]}
          keyExtractor={[Function]}
          onContentSizeChange={[Function]}
          onLayout={[Function]}
          onMomentumScrollBegin={[Function]}
          onMomentumScrollEnd={[Function]}
          onScroll={[Function]}
          onScrollBeginDrag={[Function]}
          onScrollEndDrag={[Function]}
          removeClippedSubviews={false}
          renderItem={[Function]}
          scrollEnabled={true}
          scrollEventThrottle={50}
          showsHorizontalScrollIndicator={false}
          showsVerticalScrollIndicator={false}
          stickyHeaderIndices={Array []}
          testID="activity-activity-list-index-flat-list"
          viewabilityConfigCallbackPairs={Array []}
        >
          <View
            onLayout={[Function]}
            style={null}
          >
            <View
              style={
                Object {
                  "flexDirection": "row",
                }
              }
            >
              <View
                style={
                  Object {
                    "flexDirection": "row",
                  }
                }
                testID="activity-activity-list-content-container"
              >
                <View
                  accessible={true}
                  collapsable={false}
                  focusable={false}
                  onClick={[Function]}
                  onResponderGrant={[Function]}
                  onResponderMove={[Function]}
                  onResponderRelease={[Function]}
                  onResponderTerminate={[Function]}
                  onResponderTerminationRequest={[Function]}
                  onStartShouldSetResponder={[Function]}
                  style={
                    Object {
                      "borderRadius": 10,
                      "borderWidth": 1,
                      "marginHorizontal": 5,
                      "marginVertical": 10,
                      "opacity": 1,
                      "padding": 5,
                    }
                  }
                  testID="activity-activity-list-content-button"
                >
                  <Text>
                    Content 1
                  </Text>
                </View>
              </View>
              <View
                style={
                  Object {
                    "flexDirection": "row",
                  }
                }
                testID="activity-activity-list-content-container"
              >
                <View
                  accessible={true}
                  collapsable={false}
                  focusable={false}
                  onClick={[Function]}
                  onResponderGrant={[Function]}
                  onResponderMove={[Function]}
                  onResponderRelease={[Function]}
                  onResponderTerminate={[Function]}
                  onResponderTerminationRequest={[Function]}
                  onStartShouldSetResponder={[Function]}
                  style={
                    Object {
                      "borderRadius": 10,
                      "borderWidth": 1,
                      "marginHorizontal": 5,
                      "marginVertical": 10,
                      "opacity": 1,
                      "padding": 5,
                    }
                  }
                  testID="activity-activity-list-content-button"
                >
                  <Text>
                    Content 2
                  </Text>
                </View>
              </View>
              <View
                style={
                  Object {
                    "flexDirection": "row",
                  }
                }
                testID="activity-activity-list-content-container"
              >
                <View
                  accessible={true}
                  collapsable={false}
                  focusable={false}
                  onClick={[Function]}
                  onResponderGrant={[Function]}
                  onResponderMove={[Function]}
                  onResponderRelease={[Function]}
                  onResponderTerminate={[Function]}
                  onResponderTerminationRequest={[Function]}
                  onStartShouldSetResponder={[Function]}
                  style={
                    Object {
                      "borderRadius": 10,
                      "borderWidth": 1,
                      "marginHorizontal": 5,
                      "marginVertical": 10,
                      "opacity": 1,
                      "padding": 5,
                    }
                  }
                  testID="activity-activity-list-content-button"
                >
                  <Text>
                    Content 3
                  </Text>
                </View>
              </View>
            </View>
          </View>
          <View
            onLayout={[Function]}
            style={null}
          >
            <View
              style={
                Object {
                  "flexDirection": "row",
                }
              }
            >
              <View
                style={
                  Object {
                    "flexDirection": "row",
                  }
                }
                testID="activity-activity-list-content-container"
              >
                <View
                  accessible={true}
                  collapsable={false}
                  focusable={false}
                  onClick={[Function]}
                  onResponderGrant={[Function]}
                  onResponderMove={[Function]}
                  onResponderRelease={[Function]}
                  onResponderTerminate={[Function]}
                  onResponderTerminationRequest={[Function]}
                  onStartShouldSetResponder={[Function]}
                  style={
                    Object {
                      "borderRadius": 10,
                      "borderWidth": 1,
                      "marginHorizontal": 5,
                      "marginVertical": 10,
                      "opacity": 1,
                      "padding": 5,
                    }
                  }
                  testID="activity-activity-list-content-button"
                >
                  <Text>
                    Content 4
                  </Text>
                </View>
              </View>
              <View
                style={
                  Object {
                    "flexDirection": "row",
                  }
                }
                testID="activity-activity-list-content-container"
              >
                <View
                  accessible={true}
                  collapsable={false}
                  focusable={false}
                  onClick={[Function]}
                  onResponderGrant={[Function]}
                  onResponderMove={[Function]}
                  onResponderRelease={[Function]}
                  onResponderTerminate={[Function]}
                  onResponderTerminationRequest={[Function]}
                  onStartShouldSetResponder={[Function]}
                  style={
                    Object {
                      "borderRadius": 10,
                      "borderWidth": 1,
                      "marginHorizontal": 5,
                      "marginVertical": 10,
                      "opacity": 1,
                      "padding": 5,
                    }
                  }
                  testID="activity-activity-list-content-button"
                >
                  <Text>
                    Content 5
                  </Text>
                </View>
              </View>
              <View
                style={
                  Object {
                    "flexDirection": "row",
                  }
                }
                testID="activity-activity-list-content-container"
              >
                <View
                  accessible={true}
                  collapsable={false}
                  focusable={false}
                  onClick={[Function]}
                  onResponderGrant={[Function]}
                  onResponderMove={[Function]}
                  onResponderRelease={[Function]}
                  onResponderTerminate={[Function]}
                  onResponderTerminationRequest={[Function]}
                  onStartShouldSetResponder={[Function]}
                  style={
                    Object {
                      "borderRadius": 10,
                      "borderWidth": 1,
                      "marginHorizontal": 5,
                      "marginVertical": 10,
                      "opacity": 1,
                      "padding": 5,
                    }
                  }
                  testID="activity-activity-list-content-button"
                >
                  <Text>
                    Content 6
                  </Text>
                </View>
              </View>
            </View>
          </View>
        </View>
      </View>
    </View>
  </RCTScrollView>,
]
`;
